AK/Optional.h

```diff
@@ -73,110 +98,118 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
     Optional(Optional const& other)
     requires(!IsCopyConstructible<T>)
     = delete;
-    Optional(Optional const& other) = default;
-
-    Optional(Optional&& other)
-    requires(!IsMoveConstructible<T>)
-    = delete;
-
     Optional& operator=(Optional const&)
     requires(!IsCopyConstructible<T> || !IsDestructible<T>)
     = delete;
-    Optional& operator=(Optional const&) = default;
-
-    Optional& operator=(Optional&& other)
-    requires(!IsMoveConstructible<T> || !IsDestructible<T>)
-    = delete;
 
-    ~Optional()
-    requires(!IsDestructible<T>)
-    = delete;
-    ~Optional() = default;
-
-    ALWAYS_INLINE Optional(Optional const& other)
+    constexpr ALWAYS_INLINE Optional(Optional const& other)
     requires(!IsTriviallyCopyConstructible<T>)
         : m_has_value(other.m_has_value)
     {
         if (other.has_value())
-            new (&m_storage) T(other.value());
+            construct_at<RemoveConst<T>>(&m_storage, other.value());
+        else
+            construct_null();
     }
+    constexpr ALWAYS_INLINE Optional& operator=(Optional const& other)
+    requires(!IsTriviallyCopyAssignable<T> || !IsTriviallyDestructible<T>)
+    {
+        if (this == &other)
+            return *this;
 
-    ALWAYS_INLINE Optional(Optional&& other)
+        clear();
+        m_has_value = other.m_has_value;
+        if (other.has_value())
+            construct_at<RemoveConst<T>>(&m_storage, other.value());
+        else
+            construct_null();
```

Comment:
these branches are unnecessary, assuming you don't touch `m_null` aside from actually constructing it.